### DISCUSSION TITLE
Linearization of inheritance graph impossible

### DISCUSSION THREAD

**Discussion by Lukelr12** [Cyfrin/foundry-full-course-cu#2911] [Q&A]
while following along with patrick on in lesson 20 of the stablecoin section of the course I have encountered this problem "Linearization of inheritance graph impossible". 
![Screenshot 2024-11-23 002921](https://github.com/user-attachments/assets/53c0a926-0963-41f8-ad7a-c534fc755880)

the github copilot suggestion is write a bunch of codes to basically skip the StdInvariant portion of the testing.

    '''// Override functions to resolve conflicts
    function excludeArtifact(string memory artifact) public override(StdInvariant, Test) {
        StdInvariant.excludeArtifact(artifact);
    }

    function excludeArtifacts(string[] memory artifacts) public override(StdInvariant, Test) {
        StdInvariant.excludeArtifacts(artifacts);
    }

    function excludeContract(address contractAddress) public override(StdInvariant, Test) {
        StdInvariant.excludeContract(contractAddress);
    }

    function excludeContracts(address[] memory contractAddresses) public override(StdInvariant, Test) {
        StdInvariant.excludeContracts(contractAddresses);
    }

    function excludeSelectors(bytes4[] memory selectors) public override(StdInvariant, Test) {
        StdInvariant.excludeSelectors(selectors);
    }

    function excludeSender(address sender) public override(StdInvariant, Test) {
        StdInvariant.excludeSender(sender);
    }

    function excludeSenders(address[] memory senders) public override(StdInvariant, Test) {
        StdInvariant.excludeSenders(senders);
    }

    function targetArtifact(string memory artifact) public override(StdInvariant, Test) {
        StdInvariant.targetArtifact(artifact);
    }

    function targetArtifactSelectors(string memory artifact, bytes4[] memory selectors) public override(StdInvariant, Test) {
        StdInvariant.targetArtifactSelectors(artifact, selectors);
    }

    function targetArtifacts(string[] memory artifacts) public override(StdInvariant, Test) {
        StdInvariant.targetArtifacts(artifacts);
    }

    function targetContract(address contractAddress) public override(StdInvariant, Test) {
        StdInvariant.targetContract(contractAddress);
    }

    function targetContracts(address[] memory contractAddresses) public override(StdInvariant, Test) {
        StdInvariant.targetContracts(contractAddresses);
    }

    function targetInterfaces(address contractAddress, bytes4[] memory selectors) public override(StdInvariant, Test) {
        StdInvariant.targetInterfaces(contractAddress, selectors);
    }

    function targetSelectors(bytes4[] memory selectors) public override(StdInvariant, Test) {
        StdInvariant.targetSelectors(selectors);
    }

    function targetSender(address sender) public override(StdInvariant, Test) {
        StdInvariant.targetSender(sender);
    }

    function targetSenders(address[] memory senders) public override(StdInvariant, Test) {
        StdInvariant.targetSenders(senders);
    }
}'''  

any help would be a welcome.

**Comment by Lukelr12** [comment]
answered thank you

**Reply by EngrPips** [reply]
Awesome, I am glad you figured this out.
